common/app/controllers/EmailSignupController.scala

```diff
@@ -79,6 +100,43 @@ class EmailFormService(wsClient: WSClient, emailEmbedAgent: NewsletterSignupAgen
     }
   }
 
+  def submitWithMany(form: EmailFormManyNewsletters)(implicit request: Request[AnyContent]): Future[WSResponse] = {
+    if (form.isLikelyBotSubmission) {
+      Future.failed(new IllegalAccessException("Form was likely submitted by a bot."))
```

Comment:
`IllegalAccessException` tends to be used by Java for very specific circumstances, not as a response to user input! Maybe we could create a custom exception class for this?

```
class BotSubmissionException extends Exception
```

Reply:
Because the exception is just encoding a state, you don't need a stack trace to be created which is performance costly, so you should extend [NoStackTrace](https://www.scala-lang.org/api/2.13.11/scala/util/control/NoStackTrace.html)    
```scala
class BotSubmissionException extends NoStackTrace
```

Reply:
Good to know - unless we get rid of the bot detection code that uses the error (see previous comment), I'll make that change.